modules/apps/portal-security/portal-security-content-security-policy-impl/src/main/java/com/liferay/portal/security/content/security/policy/internal/servlet/filter/ContentSecurityPolicyFilter.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.security.content.security.policy.internal.servlet.filter;
+
+import com.liferay.petra.reflect.ReflectionUtil;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.module.configuration.ConfigurationProvider;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.security.content.security.policy.configuration.ContentSecurityPolicyConfiguration;
+import com.liferay.portal.security.content.security.policy.constants.ContentSecurityPolicyConstants;
+import com.liferay.portal.servlet.filters.BasePortalFilter;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Olivér Kecskeméty
+ */
+@Component(
+	property = {
+		"after-filter=Portal CORS Servlet Filter", "dispatcher=FORWARD",
+		"dispatcher=REQUEST", "servlet-context-name=",
+		"servlet-filter-name=Content Security Policy Filter", "url-pattern=/*"
+	},
+	service = Filter.class
+)
+public class ContentSecurityPolicyFilter extends BasePortalFilter {
+
+	@Override
+	public boolean isFilterEnabled(
+		HttpServletRequest httpServletRequest,
+		HttpServletResponse httpServletResponse) {
+
+		ContentSecurityPolicyConfiguration contentSecurityPolicyConfiguration =
+			_getContentSecurityPolicyConfiguration(httpServletRequest);
+
+		return contentSecurityPolicyConfiguration.enabled();
+	}
+
+	@Override
+	protected void processFilter(
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse, FilterChain filterChain)
+		throws Exception {
+
+		ContentSecurityPolicyConfiguration contentSecurityPolicyConfiguration =
+			_getContentSecurityPolicyConfiguration(httpServletRequest);
+
+		String contentSecurityPolicy =
+			contentSecurityPolicyConfiguration.policy();
```

Comment:
Think this variable should be called "policy" to mirror the method name used to get it.